books/data_transformation.py

```diff
@@ -1,108 +1,128 @@
-        'index', 'gender', 'email', 'phone', 'cell', 'nat', 'name.title',
-        'name.first', 'name.last', 'location.street.number',
-        'location.street.name', 'location.state', 'location.postcode',
-        'location.coordinates.latitude', 'location.coordinates.longitude',
-        'location.timezone.offset', 'location.timezone.description',
-        'login.uuid', 'login.username', 'login.password', 'login.salt',
-        'login.md5', 'login.sha1', 'login.sha256', 'dob.date', 'dob.age',
-        'registered.date', 'id.name', 'id.value',
-        'picture.large', 'picture.medium', 'picture.thumbnail'
-    ])
-
-    new_api_users_column_names = {
-        'location.city': 'user_city',
-        'location.country': 'user_country',
-        'registered.age': 'user_age'
-    }
-    df_users.rename(columns=new_api_users_column_names, inplace=True)
-    df_users['user_age'] = df_users['user_age'].fillna(0).astype(int)
-    df_users.to_csv(os.path.join(load_path, 'user_api_data_transformation_v1.csv'), sep=';', index=False, header=True)
-
-
-def transform_scrapped_books_data():
-    scrapped_data = pd.read_csv(os.path.join(raw_data_path, 'scrapped_book_data.csv'), delimiter=';', usecols=['Title', 'Author', 'ISBN/ISSN:', 'Wydawnictwo:', 'Rok wydania:'], low_memory=False)
-    new_scraped_data_column_names = {
-        'ISBN/ISSN:': 'book_isbn',
-        'Title': 'book_title',
-        'Author': 'book_author',
-        'Rok wydania:': 'year_published',
-        'Wydawnictwo:': 'publisher',
-    }
-    scrapped_data.rename(columns=new_scraped_data_column_names, inplace=True)
-    scrapped_data['year_published'] = scrapped_data['year_published'].fillna(0).astype(int)
-    scrapped_data.to_csv(os.path.join(load_path, 'scrapped_books_data_transformation_v1.csv'), sep=';', index=False, header=True)
+import logging
+from argparse import ArgumentParser
+from datetime import datetime
+
+
+logging.basicConfig(filename='api.log', level=logging.INFO,
+                    format='%(asctime)s:%(levelname)s:%(message)s')
+
+
+class DataTransformation:
+
+    def __init__(self):
+        self.data_frames = {}
+        logging.info("DataTransformation initialized")
+
+    def transform_books(self, source_path):
+        books_frame = pd.read_csv(os.path.join(source_path, 'books.csv'), delimiter=';',
+                                  usecols=['ISBN', 'Book-Title', 'Book-Author', 'Year-Of-Publication', 'Publisher', 'Image-URL-S'],
+                                  encoding='windows-1250', low_memory=False)
+
+        new_books_column_names = {
+            'ISBN': 'book_isbn',
+            'Book-Title': 'book_title',
+            'Book-Author': 'book_author',
+            'Year-Of-Publication': 'year_published',
+            'Publisher': 'publisher',
+            'Image-URL-S': 'book_image_url'
+        }
+        books_frame.rename(columns=new_books_column_names, inplace=True)
+        self.data_frames['books'] = books_frame
```

Comment:
this should not be here at all. Just save the results to the target file here :)